derp/derp_server.go

```diff
@@ -508,6 +540,18 @@ func (s *Server) accept(nc Conn, brw *bufio.ReadWriter, remoteAddr string, connN
 	return c.run(ctx)
 }
 
+// toKB converts bytes to kilobytes, rounding up if necessary.
+func toKB(bytes uint32) uint32 { return (bytes + (1 << 10) - 1) >> 10 }
+
+func (c *sclient) rateLimit(kb uint32) {
+	// TODO do we want to have some timeout for things here?
+	ctx := context.TODO()
+	if c.limiter != nil {
```

Comment:
you documented that if this is nil, rate limit isn't enabled, but you are on line 552 below, which is an added mutex that all our prod traffic would be hitting even when don't want it (we rate-limit at the OS level)

Reply:
Hm, there are two limiters here, one on the server and one on the clients. I had the server rate limiting always be on, but I moved it to a separate function and documented the nil behavior.